ports/esp32/esp32_rmt.c

```diff
@@ -160,63 +160,107 @@ static mp_obj_t esp32_rmt_make_new(const mp_obj_type_t *type, size_t n_args, siz
             mp_raise_ValueError(MP_ERROR_TEXT("tx_carrier duty must be 0..100"));
         }
 
-        config.tx_config.carrier_en = 1;
-        config.tx_config.carrier_freq_hz = frequency;
-        config.tx_config.carrier_duty_percent = duty;
-        config.tx_config.carrier_level = level;
-    } else {
-        config.tx_config.carrier_en = 0;
+        rmt_carrier_config_t tx_carrier_cfg = {
+            .duty_cycle = ((float)duty) / 100.0,
+            .frequency_hz = frequency,
+            .flags.polarity_active_low = !level,
+        };
+        check_esp_err(rmt_apply_carrier(self->channel, &tx_carrier_cfg));
     }
 
-    config.tx_config.idle_output_en = 1;
-    config.tx_config.idle_level = idle_level;
-
-    config.clk_div = self->clock_div;
+    rmt_copy_encoder_config_t copy_encoder_config = {};
+    check_esp_err(rmt_new_copy_encoder(&copy_encoder_config, &self->encoder));
 
-    check_esp_err(rmt_config(&config));
-    check_esp_err(rmt_driver_install_core1(config.channel));
+    rmt_tx_event_callbacks_t callbacks = {
+        .on_trans_done = esp32_rmt_tx_trans_done,
+    };
+    check_esp_err(rmt_tx_register_event_callbacks(self->channel, &callbacks, self));
 
     return MP_OBJ_FROM_PTR(self);
 }
 
 static void esp32_rmt_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
     esp32_rmt_obj_t *self = MP_OBJ_TO_PTR(self_in);
     if (self->pin != -1) {
-        bool idle_output_en;
-        rmt_idle_level_t idle_level;
-        check_esp_err(rmt_get_idle_level(self->channel_id, &idle_output_en, &idle_level));
-        mp_printf(print, "RMT(channel=%u, pin=%u, source_freq=%u, clock_div=%u, idle_level=%u)",
-            self->channel_id, self->pin, APB_CLK_FREQ, self->clock_div, idle_level);
+        mp_printf(print, "RMT(pin=%u, source_freq=%u, resolution_hz=%u, idle_level=%u)",
+            self->pin, self->clock_freq, self->resolution_hz, self->idle_level);
     } else {
         mp_printf(print, "RMT()");
     }
 }
 
+static void esp32_rmt_deactivate(esp32_rmt_obj_t *self) {
+    if (self->enabled) {
+        // FIXME: panics in ESP32 if called while TX is ongoing and TX sequence is long (>300ms)
+        // Does not panic in ESP32-S3, ESP32-C3 and ESP32-C6.
+        // Tested with ESP-IDF up to 5.5
+        // ESP-IDF issue: https://github.com/espressif/esp-idf/issues/17692
+        //
+        // Cause is Interrupt WDT to trigger because ESP-IDF rmt_disable() disables
+        // interrupts and spinlocks until the ongoing TX sequence is finished.
+        //
+        // Workaround is never try to stop RMT sequences longer than 300ms (which are unusual
+        // anyway). Or apply the patch mentioned at the GitHub issue to ESP-IDF.
+        rmt_disable(self->channel);
+        self->enabled = false;
+    }
+}
+
+static mp_obj_t esp32_rmt_active(size_t n_args, const mp_obj_t *args) {
```

Comment:
Is this method needed?  It's not added to the documentation, so it's not clear exactly what it's used for.  Eg what's the difference between `active(False)` and `deinit()`?

Also, this method can't be used to activate the RMT, so that's another reason why maybe it's not needed at all.